### PR TITLE
USACO Jan 22 Probs

### PR DESCRIPTION
**Pull request by bqi343**
`/general/usaco-monthlies`

Last month's PR: https://github.com/cpinitiative/usaco-guide/pull/2433/files

Will wait for one or two approvals.

_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

**Comment by SansPapyrus683** [comment]
what about gold p3

**Comment by bqi343** [comment]
Added.

**Comment by SansPapyrus683** [comment]
some problems from bronze & silver are missing as well, is this intentional

**Comment by bqi343** [comment]
I didn't know what to tag silver problem #1 with. anyways, it's not necessary to add tags for all the problems

**Comment by SansPapyrus683** [comment]
oh silver p1 is still going to be in there?

**Comment by bqi343** [comment]
Yes, you can check the link.

File: content/extraProblems.json [hunk 1]
```diff
@@ -589,6 +589,149 @@
...
+      }
+    },
+    {
+      "uniqueId": "usaco-1186",
+      "name": "Farm Updates",
+      "url": "http://www.usaco.org/index.php?page=viewproblem2&cpid=1186",
+      "source": "Gold",
+      "difficulty": "Hard",
+      "isStarred": false,
+      "tags": ["Connected Components"],
```
Comment (SansPapyrus683) [inline content/extraProblems.json:677]: dsu maybe

**Review by SansPapyrus683 — APPROVED** [review]

File: content/extraProblems.json (hunk 1 above)
Comment (bqi343) [reply content/extraProblems.json:677]: The official solution doesn't use it.

Comment (SansPapyrus683) [reply content/extraProblems.json:677]: so? that doesn't mean you can't solve it w/ dsu?

Comment (bqi343) [reply content/extraProblems.json:677]: I don't see a reason to tag it with DSU if DSU isn't necessary to solve it and doesn't make the solution any simpler.

Comment (SansPapyrus683) [reply content/extraProblems.json:677]: so what you're saying is alternative methods should only be mentioned if they make the solution simpler?

Comment (bqi343) [reply content/extraProblems.json:677]: Yes.

1. I do not see how a solution involving DSU would be simpler than the solution in the official analysis.
2. If the only criterion for tagging a problem with DSU was if DSU could potentially be used, then (almost?) every problem involving connected components would also be tagged with DSU.
3. DSU is considered a Gold-level topic. By not tagging it with DSU, I hope I made it clear that this problem could very well have been used in Silver (in place of Cereal 2).

Comment (SansPapyrus683) [reply content/extraProblems.json:677]: lmao sure

Comment (SansPapyrus683) [reply content/extraProblems.json:677]: i mean i still disagree but you make some good points
the pr's merged anyways lol

File: content/extraProblems.json [hunk 2]
```diff
@@ -589,6 +589,149 @@
...
+      }
+    },
+    {
+      "uniqueId": "usaco-1184",
+      "name": "Cereal 2",
+      "url": "http://www.usaco.org/index.php?page=viewproblem2&cpid=1184",
+      "source": "Silver",
+      "difficulty": "Very Hard",
+      "isStarred": false,
+      "tags": ["Spanning Tree", "Constructive"],
```
Comment (andrewwangva) [inline content/extraProblems.json:651]: I know this is late - but connected components? Did not see an official solution posted yet so idk what your sol. was.

Comment (bqi343) [reply content/extraProblems.json:651]: check the analysis. of course, you'll need to find the connected components before finding a spanning tree for each component

**Pull request merged**